hazelcast-jet-core/src/main/java/com/hazelcast/jet/Processors.java

```diff
@@ -375,11 +377,13 @@ public static ProcessorSupplier streamFiles(@Nonnull String watchedDirectory) {
      *
      * @param watchedDirectory The directory where we watch files
      * @param charset charset to use to decode the files, or {@code null} to use UTF-8
+     * @param glob the file filter, see {@link
+     *             java.nio.file.FileSystem#getPathMatcher(String) getPathMatcher()}
      */
     public static ProcessorSupplier streamFiles(
-            @Nonnull String watchedDirectory, @Nullable Charset charset
+            @Nonnull String watchedDirectory, @Nullable Charset charset, @Nullable String glob
     ) {
-        return StreamFilesP.supplier(watchedDirectory, charset == null ? null : charset.toString());
+        return StreamFilesP.supplier(watchedDirectory, charset == null ? null : charset.toString(), glob);
```

Comment:
For me, "null" meaning "use default" is perfectly acceptable. Especially here, when you might want to use just one of the parameters (either glob or charset) and want the other to be the default. You just don't care. I didn't want to add another 2 overloads... Yesterday, it also made the test more complicated because I cannot say "use default" using parameters; I had to use different overload to test the default.